plugins/com.google.cloud.tools.eclipse.appengine.compat/src/com/google/cloud/tools/eclipse/appengine/compat/GpeMigrator.java

```diff
@@ -108,29 +120,35 @@ private static boolean isGpeClasspath(IClasspathEntry entry) {
   }
 
   @VisibleForTesting
-  static void removeGpeNature(IProject project) {
+  static boolean removeGpeNature(IProject project) {
     try {
+      boolean hadNature = NatureUtils.hasNature(project, GPE_GAE_NATURE_ID);
       NatureUtils.removeNature(project, GPE_GAE_NATURE_ID);
+      return hadNature;
     } catch (CoreException ex) {
       logger.log(Level.WARNING, "Failed to remove GPE nature.", ex);
+      return true;
```

Comment:
I'll surface the `CoreException` upwards: it's only thrown if the project doesn't exist, isn't open, or has some other major issue. In which case there's not much we can do: the project itself is unusable.